pycbc/waveform/waveform_modes.py

```diff
@@ -0,0 +1,491 @@
+        hneg = FrequencySeries(hneg.data.data, delta_f=hneg.deltaF,
+                               epoch=hneg.epoch)
+        if return_posneg:
+            hlms[ell, m] = (hpos, hneg)
+        else:
+            # convert to ulm, vlm
+            ulm = 0.5 * (hpos + hneg.conj())
+            vlm = 0.5j * (hneg.conj() - hpos)
+            hlms[ell, m] = (ulm, vlm)
+    return hlms
+
+
+def phenom_l0frame_to_jframe(approximant, mass1, mass2, f_ref, phiref=0.,
+                             inclination=0.,
+                             spin1x=0., spin1y=0., spin1z=0.,
+                             spin2x=0., spin2y=0., spin2z=0.):
+    r"""Converts L0- to J-frame parameters used by IMRPhenomP waveforms.
+
+    Parameters
+    ----------
+    approximant : str
+        Name of the approximant. Must be one of the IMRPhenom approximants.
+    {mass1}
+    {mass2}
+    {f_ref}
+    phiref : float
+        Reference phase.
+    inclination : float
+        Angle between the orbital angular momentum at ``f_ref`` and the line
+        of sight.
+    {spin1x}
+    {spin1y}
+    {spin1z}
+    {spin2x}
+    {spin2y}
+    {spin2z}
+
+    Returns
+    -------
+    dict :
+        Dictionary of:
+        * thetajn : float
+            Angle between the line of sight and the total angular momentum.
+            This is the thing that goes into the polar angle part of the
+            spherical harmonics.
+        * alpha0 : float
+            Azimuthal angle in the J frame. This is the thing that goes into
+            the azimuthal part of the spherical harmonics.
+        * phi_aligned : float
+            Beats me.
+        * zeta_polarization : float
+            Another mystery.
+        * spin1_l : float
+            Component of the larger object's spin that is aligned with the
+            orbital angular momentum.
+        * spin2_l : float
+            Component of the smaller object's spin that is aligned with the
+            orbital angular momentum.
+        * {chi_p}
+    """
+    phenomv = approximant.replace('HM', '') + '_V'
+    spin1_l, spin2_l, chip, thetajn, alpha0, phi_aligned, zeta_pol = \
+        lalsimulation.SimIMRPhenomPCalculateModelParametersFromSourceFrame(
+            mass1*lal.MSUN_SI, mass2*lal.MSUN_SI, f_ref, phiref, inclination,
+            spin1x, spin1y, spin1z, spin2x, spin2y, spin2z,
+            getattr(lalsimulation, phenomv))
+    out = {'thetajn': thetajn,
+           'alpha0': alpha0,
+           'phi_aligned': phi_aligned,
+           'zeta_polarization': zeta_pol,
+           'spin1_l': spin1_l,
+           'spin2_l': spin2_l,
+           'chi_p': chip}
+    return out
+
+
+phenom_l0frame_to_jframe.__doc__ = _formatdocstr(
+    phenom_l0frame_to_jframe.__doc__)
+
+
+def l0frame_to_jframe(mass1, mass2, f_ref, phiref=0., inclination=0.,
+                      spin1x=0., spin1y=0., spin1z=0.,
+                      spin2x=0., spin2y=0., spin2z=0.):
+    """Converts L0-frame parameters to J-frame.
```

Comment:
If Phenom is somehow using something different (which is a problem, as lalsimulation does not allow for this freedom), how will the user know that this function won't work?

Reply:
🤷‍♂️ I'm still not certain on what the different conventions are for the different approximants. But the `J -> L0` frame conversion should be applicable to the frame that `ChooseFDWaveform` and `ChooseTDWaveform` use. That function is actually what they use in LALInference and Bilby to convert inclination into thetaJN. It's just that when you start digging down and trying to call individual approximant's functions that you get into the weeds of trying to figure out what convention a particular waveform uses. Anyway, I ended up not actually using these functions in this module. I decided to leave them in though since we currently don't have a function in pycbc that will do the conversion between inclination and thetaJN. In another PR I'll probably add a wrapper function specifically for that in `conversions.py`.

Reply:
Okay, I'm happy with the `l0frame_to_jframe` function but should we really keep a `phenom_l0frame_to_jframe` function as well? My take is that if the waveform is wrong, it should be fixed!

Reply:
Fair enough; removed.